hbp_validation_framework/__init__.py

```diff
@@ -338,10 +344,12 @@ def _download_resource(self, uuid):
             raise Exception("The provided 'uuid' is invalid!")
         else:
             data = data.json()
+            dirpath = target_loc
             if data["entity_type"] == "folder":
-                if not os.path.exists(data["name"]):
-                    os.makedirs(data["name"])
-                os.chdir(data["name"])
+                dirpath = os.path.join(target_loc, data["name"])
+                if not os.path.exists(dirpath):
```

Comment:
maybe use the `Path` module so as to support the `exists_ok` option. (what if `target_loc` is a deeply nested path where some of the intermediate directories don't exist?)

Reply:
I think `os.makedirs` takes care of any missing intermediate directories:
https://docs.python.org/2/library/os.html#os.makedirs

But yes the `Path` module seems more appropriate for this purpose. Will shift to this.

Reply:
Sorry, I mixed things up. The problem with `os.makedirs` is when the final directory already exists, but this is handled by your `os.path.exists` call

Reply:
Decided to implement all directory creation using `Path` module. Changes made accordingly.